### PR TITLE
Add additional details to implementation plan template

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description

When reviewing https://github.com/WordPress/openverse/pull/2676 I realised that our implementation plan template is missing what I view as a key section: the step-by-step plan. An ordered list of the steps is so useful for getting a high-level understanding of how the different requirements of the plan fit together and how the implementation will be paced.

I added a new section for this and renamed "outlined steps" to "step details" to further help disambiguation between the new section and the existing one.

I also added a new section for feature flags, because that was also not explicitly required in the implementation plan despite being a critical consideration.

## Testing Instructions

Read the updated template and the description comments I've added or expanded upon and confirm they clearly describe the intention of each section.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [N/A] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: documentation/projects/templates/implementation-plan.md [hunk 1]
```diff
@@ -30,12 +30,34 @@
...
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+## Step details
+
+<!--
+Describe the implementation step necessary for completion.
```
Comment (obulat) [inline documentation/projects/templates/implementation-plan.md:49]:
```suggestion
Describe all of the implementation steps listed in the "step-by-step plan" in detail.
```

**Review by obulat — APPROVED** [review]
Thank you for adding this! I was trying to do both the detailed description and the plan in the same section (in different plans), and it was confusing.

File: documentation/projects/templates/implementation-plan.md [hunk 2]
```diff
@@ -30,12 +30,34 @@
 
 <!-- List any succinct expected products from this implementation plan. -->
 
-## Outlined Steps
+## Step-by-step plan
 
-<!-- Describe the implementation step necessary for completion. -->
+<!--
+List the ordered steps of the plan in the form of imperative-tone issue titles.
```
Comment (zackkrida) [inline documentation/projects/templates/implementation-plan.md:36]: 👍

**Review by zackkrida — APPROVED** [review]
Great addition!

**Pull request merged**